llvm/lib/MC/MCInstPrinter.cpp

```diff
@@ -226,27 +226,32 @@ format_object<uint64_t> MCInstPrinter::formatHex(uint64_t Value) const {
 
 MCInstPrinter::WithMarkup MCInstPrinter::markup(raw_ostream &OS,
                                                 Markup S) const {
-  return WithMarkup(OS, S, getUseMarkup(), getUseColor());
+  return WithMarkup(const_cast<MCInstPrinter &>(*this), OS, S, getUseMarkup(),
```

Comment:
Overall approach looks fine, but this does feel a bit like an abuse of `const`. Could `const` be dropped from the `markup` method?

Reply:
Many const methods transitively call the `const` `markup`. If we want to drop `const_cast` here, many target functions `printXXX` will need be adjusted...

Reply:
Yuck, okay. I think that's probably still the "right" thing to do (after all you're modifying the state of the class when calling these methods in this manner). But I acknowledge that's probably a bit impractical too, so if you add a TODO comment here instead, I'd be happy with that.

I also considered making the `ColorStack` member `mutable`, but that doesn't really feel any better than the `const_cast` in this context.

Reply:
Migrated `printRegName` users. Dropped `const_cast` now.